sentry_sdk/client.py

```diff
@@ -466,20 +466,28 @@ def _should_sample_error(
         hint,  # type: Hint
     ):
         # type: (...) -> bool
-        sampler = self.options.get("error_sampler", None)
+        error_sampler = self.options.get("error_sampler", None)
 
-        if callable(sampler):
+        if callable(error_sampler):
```

Comment:
```suggestion
        if error_sampler is not None:
```
I think we should check that the `error_sampler` is not None here rather than check whether it is callable. This way, if the user supplies an `error_sampler` that cannot be called, we handle it in the same way as if the `error_sampler` raises an error when called

Reply:
This would change existing behavior. Up to now if someone submitted a string as `error_sampler` it took the `sample_rate` from the options. With this change we would not take the `sample_rate` from options but default to always sampling the event. We should not change existing behavior.

Reply:
Good point -- this is probably an edge case anyways, so I am fine with this change. Will approve the PR